src/components/MachineSelector/machineSelector.test.tsx

```diff
@@ -0,0 +1,57 @@
+import * as React from "react";
+import axios from "axios";
+import { waitFor, fireEvent } from "@testing-library/react";
+import MachineSelector from ".";
+import { render } from "../../utils/customRender";
+import { getActiveMachines } from "../../services/getActiveMachines";
+import { mockMachines } from "../../mock/mockMachines";
+
+jest.mock("axios");
+const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+describe("tests for the MachineSelector Component", () => {
+  beforeEach(() => {
+    mockedAxios.get.mockReset();
+  });
+
+  test("renders correctly", async () => {
+    const { getByTestId } = render(<MachineSelector />);
+    const selectElement = await waitFor(() => getByTestId("machine-selector"));
+
+    expect(selectElement).toBeTruthy();
+  });
+
+  test("initially renders with localhost ", async () => {
+    mockedAxios.get.mockResolvedValue({ data: mockMachines });
+
+    const { getByTestId } = render(<MachineSelector />);
+    const selectElement = await waitFor(() => getByTestId("machine-selector"));
+
+    expect(selectElement).toHaveValue("localhost");
+    expect(mockedAxios.get).toHaveBeenCalledWith(getActiveMachines());
+  });
+
+  test("changes value on selecting different option ", async () => {
+    mockedAxios.get.mockResolvedValue({ data: mockMachines });
+
+    const { getByTestId } = render(<MachineSelector />);
```

Comment:
@aquibbaig Is there any possibility we can declare the constant `getByTestId` globally in line number 16.

Reply:
I dont think so @muskankhedia actually after every test cleanup is run automatically by rtl so that we always start with a clean slate in every test

Reply:
Okay, got it.